### PR TITLE
Initial support for local notifications!

### PR DESCRIPTION
**Pull request by JBassett**
## Summary

Like iOS, this adds initial support for local (no FCM) notifications.
Items left to complete:

- [x] Change map either local or FCM message so they are the same
- [x] Verify all options still working
- [x] Create setting to choose when we listen for these on the websocket (never, screen on, all the time)?
- [x] Add support for minimal builds

## Screenshots

Screenshot_1641070536

## Link to pull request in Documentation repository

https://github.com/home-assistant/companion.home-assistant/pull/648

## Any other notes

Fixes: https://github.com/home-assistant/android/issues/1480

### CODE REVIEW

**Comment by dshokouhi** [comment]
Started running tests on my local devices here, should have testing completed tonight :) also this should close https://github.com/home-assistant/android/issues/1480 !

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -216,23 +230,22 @@ class SettingsFragment constructor(
...
-            val hasSamsungApp = pm.getLaunchIntentForPackage("com.samsung.android.app.watchmanager")
-            findPreference<PreferenceCategory>("wear_category")?.isVisible = hasWearApp != null || hasSamsungApp != null
-            findPreference<Preference>("wear_settings")?.setOnPreferenceClickListener {
-                startActivity(SettingsWearActivity.newInstance(requireContext()))
-                return@setOnPreferenceClickListener true
-            }
+        val pm = requireContext().packageManager
+        val hasWearApp = pm.getLaunchIntentForPackage("com.google.android.wearable.app")
+        val hasSamsungApp = pm.getLaunchIntentForPackage("com.samsung.android.app.watchmanager")
+        findPreference<PreferenceCategory>("wear_category")?.isVisible = hasWearApp != null || hasSamsungApp != null
```
Comment (dshokouhi) [inline app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt:245]: We should probably add a `full` check here for the Wear OS category to be hidden from minimal like before

Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt]: Fixed

**Comment by JBassett** [comment]
Updated setting to generic websocket setting.  Defaults to Screen On.

**Comment by JBassett** [comment]
Updated default for minimal builds to be always since they don't have a default way to get notifications.

File: app/src/full/java/io/homeassistant/companion/android/notifications/FirebaseCloudMessagingService.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class FirebaseCloudMessagingService : FirebaseMessagingService() {
+    companion object {
+        const val TAG = "MessagingService"
```
Comment (dshokouhi) [inline app/src/full/java/io/homeassistant/companion/android/notifications/FirebaseCloudMessagingService.kt:20]: Should we change this so we know the log came from firebase? I noticed `MessagingManager` has the same `TAG` so didnt want to confuse them :)

File: app/src/main/java/io/homeassistant/companion/android/settings/websocket/WebsocketSettingFragment.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.platform.ComposeView
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.google.android.material.composethemeadapter.MdcTheme
+import io.homeassistant.companion.android.settings.websocket.views.WebsocketSettingView
+
+// @AndroidEntryPoint
```
Comment (dshokouhi) [inline app/src/main/java/io/homeassistant/companion/android/settings/websocket/WebsocketSettingFragment.kt:14]: no longer needed? :)

File: app/src/main/java/io/homeassistant/companion/android/websocket/WebsocketManager.kt [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+class WebsocketManager(
+    appContext: Context,
+    workerParams: WorkerParameters
+) : CoroutineWorker(appContext, workerParams) {
+
+    companion object {
+        private const val TAG = "WebSockManager"
+        private const val CHANNEL_ID = "Websocket"
+        private const val NOTIFICATION_ID = 65423
+        private val DEFAULT_WEBSOCKET_SETTING = if (BuildConfig.BUILD_TYPE == "minimal") WebsocketSetting.ALWAYS else WebsocketSetting.SCREEN_ON
```
Comment (dshokouhi) [inline app/src/main/java/io/homeassistant/companion/android/websocket/WebsocketManager.kt:48]: Should we make this always only if the build is `full`? I know we cant post notifications on the `quest` build yet but I think some notifications commands may work :)

**Review by dshokouhi — COMMENTED** [review]
Loaded up the debug APK (full for now but set to `always`) also some small comments. Will report back after 24 hours or so to see if there are any disconnects.

File: app/src/main/java/io/homeassistant/companion/android/websocket/WebsocketManager.kt (hunk 4 above)
Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/websocket/WebsocketManager.kt]: Fixed.

File: app/src/main/java/io/homeassistant/companion/android/settings/websocket/WebsocketSettingFragment.kt (hunk 3 above)
Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/settings/websocket/WebsocketSettingFragment.kt]: Fixed.

File: app/src/full/java/io/homeassistant/companion/android/notifications/FirebaseCloudMessagingService.kt (hunk 2 above)
Comment (JBassett) [reply app/src/full/java/io/homeassistant/companion/android/notifications/FirebaseCloudMessagingService.kt]: Fixed.

**Review by dshokouhi — APPROVED** [review]
Ran some tests over a 24 hour period and I don't seem to see any issues. As the debug APK does not have FCM notifications I could not test the fallback to FCM so we will need to test in the beta. Beta feedback will be critical here so lets get it out for our testers so they can let us know how things are working!

**Pull request merged**